packages/plugin-nft-generation/src/provider/wallet/walletSolana.ts

```diff
@@ -0,0 +1,231 @@
+        const info = {
+            name,
+            symbol,
+            uri,
+        };
+        console.log(`Metadata uploaded: ${uri}`);
+        transaction = transaction.add(
+            createNft(this.umi, {
+                ...info,
+                mint: collectionMint,
+                sellerFeeBasisPoints: percentAmount(fee),
+                isCollection: true,
+            })
+        );
+
+        transaction = transaction.add(
+            updateV1(this.umi, {
+                mint: collectionMint.publicKey,
+                newUpdateAuthority: publicKey(adminPublicKey), // updateAuthority's public key
+            })
+        );
+
+        await transaction.sendAndConfirm(this.umi, {
+            confirm: {},
+        });
+
+        // await sleep(5000);
+        // const createdCollectionNft = await fetchDigitalAsset(
+        //     this.umi,
+        //     collectionMint.publicKey
+        // );
+        const address = collectionMint.publicKey;
+        return {
+            link: getExplorerLink("address", address, this.cluster),
+            address,
+        };
+    }
+
+    async mintNFT({
+        collectionAddress,
+        adminPublicKey,
+        name,
+        symbol,
+        uri,
+        fee,
+    }: {
+        collectionAddress: string;
+        adminPublicKey: string;
+        name: string;
+        symbol: string;
+        uri: string;
+        fee: number;
+    }): Promise<any> {
+        const umi = this.umi;
+        const mint = generateSigner(umi);
+
+        let transaction = new TransactionBuilder();
+        console.log("collectionAddress", collectionAddress);
+        const collectionAddressKey = publicKey(collectionAddress);
+        // Add SOL transfer instruction (0.1 SOL)
+        const receiverAddressKey = publicKey(adminPublicKey); // Replace with actual receiver address
+        transaction = transaction.add(
+            transferSol(umi, {
+                source: umi.identity,
+                destination: receiverAddressKey,
+                amount: sol(0.1),
+            })
+        );
+        const info = {
+            name,
+            uri,
+            symbol,
+        };
+        transaction = transaction.add(
+            createNft(umi, {
+                mint,
+                ...info,
+                sellerFeeBasisPoints: percentAmount(fee),
+                collection: {
+                    key: collectionAddressKey,
+                    verified: false,
+                },
+            })
+        );
+
+        transaction = transaction.add(
+            updateV1(umi, {
+                mint: mint.publicKey,
+                newUpdateAuthority: publicKey(adminPublicKey), // updateAuthority's public key
+            })
+        );
+
+        await transaction.sendAndConfirm(umi);
+        //
+        // await sleep();
+        // const createdNft = await fetchDigitalAsset(umi, mint.publicKey);
+        //
+        // console.log(
```

Comment:
For debugging efforts, would it be appropriate leave logs here to help inform on the status of minting the NFT? 

What happens if the mint fails? Will it halt the agent?